spec/osut_tests_spec.rb

```diff
@@ -3431,311 +4303,242 @@ module M
+    file = File.join(__dir__, "files/osms/out/office_plenum.osm")
+    model.save(file, true)
+
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # CASE 2b:
+    file  = File.join(__dir__, "files/osms/in/smalloffice.osm")
+    path  = OpenStudio::Path.new(file)
+    model = translator.loadModel(path)
+    expect(model).to_not be_empty
+    model = model.get
+
+    core  = model.getSpaceByName("Core_ZN")
+    attic = model.getSpaceByName("Attic")
+    expect(core).to_not be_empty
+    expect(attic).to_not be_empty
+    core  = core.get
+    attic = attic.get
+
+    # Again, tag attic as an INDIRECTLY-CONDITIONED space.
+    key = "indirectlyconditioned"
+    val = core.nameString
+    expect(attic.additionalProperties.setFeature(key, val)).to be true
+    expect(mod1.plenum?(attic)).to be false
+    expect(mod1.unconditioned?(attic)).to be false
+    expect(mod1.setpoints(attic)[:heating]).to be_within(TOL).of(21.11)
+    expect(mod1.setpoints(attic)[:cooling]).to be_within(TOL).of(23.89)
+    expect(mod1.status).to be_zero
+
+    gra_plenum = mod1.grossRoofArea(core)
+    expect(mod1.status).to be_zero
+    expect(gra_plenum.round(2)).to eq(total.round(2))
+
+    # Conflicting argument case: Here, the method can only add skylight wells
+    # through model "plenums" (in this context, :plenum is an all encompassing
+    # keyword for any INDIRECTLY-CONDITIONED, unoccupied space). Yet by passing
+    # option "plenum: false", the method is instructed to skip "plenum"
+    # skylight wells altogether.
+    rm2 = mod1.addSkyLights(core, {srr: srr, plenum: false})
+    expect(mod1.status).to be_zero
+    expect(rm2.round(2)).to eq(total.round(2))
+
+    core_skies = mod1.facets(core, "Outdoors", "Skylight")
+    sky_area2  = core_skies.sum(&:grossArea)
+    expect(sky_area2.round(2)).to eq(0.00)
 
-      file = File.join(__dir__, "files/osms/out/office.osm")
-      model.save(file, true)
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    file  = File.join(__dir__, "files/osms/in/warehouse.osm")
+    path  = OpenStudio::Path.new(file)
+    model = translator.loadModel(path)
+    expect(model).to_not be_empty
+    model = model.get
+
+    model.getSpaces.each do |space|
+      id = space.nameString
+      next unless space.partofTotalFloorArea
+
+      sidelit = mod1.daylit?(space, true, false)
+      toplit  = mod1.daylit?(space, false)
+      expect(sidelit).to be true  if id.include?("Office")
+      expect(sidelit).to be false if id.include?("Storage")
+      expect(toplit ).to be false if id.include?("Office")
+      expect(toplit ).to be true  if id.include?("Storage")
     end
+
+    bulk = model.getSpaceByName("Zone3 Bulk Storage")
+    fine = model.getSpaceByName("Zone2 Fine Storage")
+    expect(bulk).to_not be_empty
+    expect(fine).to_not be_empty
+    bulk = bulk.get
+    fine = fine.get
+
+    # No overhangs/attics. Calculation of roof area for SRR% is more intuitive.
```

Comment:
Models without plenums/attics (e.g. warehouses) are far easier to process (no wells).

<img width="1422" alt="warehouse" src="https://github.com/rd2/osut/assets/98104985/f8b5c41b-ac80-488f-b308-e6ebc8a425b6">